content/docs/2.17/scalers/temporal.md

```diff
@@ -0,0 +1,108 @@
+    metadata:
+      namespace: default
+      taskQueue: "workflow_with_single_noop_activity:test"
+      targetQueueSize: "2"
+      activationTargetQueueSize: "0"
+      endpoint: temporal-frontend.temporal.svc.cluster.local:7233
+```
+
+**Parameter list:**
+
+- `endpoint` - This parameter specifies the URL of the Temporal gRPC service. You need to provide the service address in the format `<hostname>:<port>`. (Required)
+- `endpointFromEnv` - Defines the endpoint, similar to the `endpoint` parameter, but the value is read from an environment variable. (Optional)
+- `namespace` - The namespace of the temporal service. (Default:`default`, Optional)
+- `activationTargetQueueSize` - This sets the target value for activating the scaler. More information about activation thresholds can be found  [here](./../concepts/scaling-deployments.md#activating-and-scaling-thresholds). (Default: `0`, Optional)
+- `targetQueueSize` - Target value for queue length passed to the scaler. The scaler will cause the replicas to increase if the queue message count is greater than the target value per active replica. (Default: `5`, Optional)
+- `taskQueue` - This parameter specifies the task queue name. (Required)
+- `queueTypes` - Task Queue type, which can be either workflow or activity. The default type is workflow (Optional)
+- `buildId` - Build IDs identify Worker versions for Workflow versioning and task compatibility (Optional)
+- `selectAllActive` - Include all active versions (Default:`false`, Optional)
+- `selectUnversioned` - Include the unversioned queue (Default:`false`, Optional)
+- `apiKey` - API key for authentication with temporal cloud (Optional)
+- `apiKeyFromEnv` - API key for authentication similer to `apiKey`, but read from an environment variable (Optional)
+- `minConnectTimeout` - This is the minimum amount of time we are willing to give a connection to complete. (Default:`5`, Optional)
+- `UnsafeSsl` - Whether to allow unsafe SSL (Default: `false`, Optional)
+
+**Authentication Parameters:**
+
+Temporal supports `apiKey` and `mTLS` for authentication. You can use the following parameters to configure authentication:
+
+- `apiKey` - API key for authentication with temporal cloud. (Optional)
+- `ca` - Certificate authority file for TLS client authentication. (Optional)
+- `cert` - Certificate for client authentication. (Optional)
+- `key` - Key for client authentication. (Optional)
+- `keyPassword` - If set the keyPassword is used to decrypt the provided key. (Optional)
+
+> 💡 **NOTE:** Activation based on backlog may not be reliable when scaling to zero.
+  This approach fails to account for in-flight tasks or workloads with throughput too low to trigger a backlog.
+  Consequently, scaling to zero could result in undesirable behavior,
+  such as terminating resources and subsequently having to scale back up to handle queued tasks. To address these challenges, consider customizing the scale-down behavior of the Horizontal Pod Autoscaler (HPA).
```

Comment:
For scale to zero, the main setting is cooldownPeriod, isn't that what we want to highlight here?

HPA behavior is for 1-N scaling. Does this issue also affects scaling in, ie for example from 5 -> 4? If so, then HPA behavior setting makes sense

Reply:
yes it affects  scaling in as well. Can we mention `cooldownPeriod` also?

Reply:
So it affects both 1->0 and N->M? If so, we should aslo mention `cooldownPeriod`

Reply:
Done